src/components/UserManagedList.js

```diff
@@ -174,8 +175,8 @@ export default class UserManagedList extends SystemManagedList {
   async doSave() {
     const { components } = this.props
     const spec = this.buildSaveSpec(components.list)
-    await this.saveSpec(spec)
     this.setState({ showSavePopup: false })
+    this.saveSpec(spec)
```

Comment:
If this method needs to be called after the state change, then is better to use the callback method of the `setState()` function:
`this.setState({ showSavePopup: false }, () => this.saveSpec(spec))`